### ISSUE TITLE
Hints on NZZ page don't work.

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#1298]
On http://www.nzz.ch/international/europa/ausnahmezustand-kommt-in-verfassung-1.18692331 when trying to click the "Kampf gegen den Terrorismus" headline under "Zum Thema" via hints, nothing happens.

Also `flipppp` needs two middle-clicks to open the link, while it seems to work fine for me.

**Comment by error800** [comment]
This seems to be a more general problem. When an <code>&lt;a&gt;</code> includes another container and not just text, hints do not work. Text formatting tags, like <code>&lt;br&gt;</code>, <code>&lt;em&gt;</code>, <code>&lt;strong&gt;</code>, <code>&lt;i&gt;</code>, <code>&lt;b&gt;</code>... work fine though.
```html
<p>this works with hints:</p>
<a href="http://qutebrowser.org">
  link<br>
  <em>link</em><br>
  <strong>link</strong><br>
  <i>link</i><br>
  <b>link</b><br>
  link
</a>

<p>this doesn't:</p>
<a href="http://qutebrowser.org">
  <p>link</p>
</a>
```

**Comment by error800** [comment]
The problem seems to occur as soon as there is a block level element inside an <code>&lt;a&gt;</code>

```html
<p>this works:</p>
<a href="http://qutebrowser.org">
  <p style="display: inline;">link</p>
</a>
<p>this doesn't:</p>
<a href="http://qutebrowser.org">
  <p>link</p>
</a>
```

**Comment by lahwaacz** [comment]
When there is a block element inside `<a>`, the inspect tool shows its size as `0px x 0px`, only the nested block element gets positive dimensions. Hence emulating a click event at the middle of `<a>` does not work as expected.

**Comment by lahwaacz** [comment]
I've been thinking about this and it seems that the trick I proposed in https://github.com/The-Compiler/qutebrowser/issues/1182#issuecomment-180372703 might work even in this case :wink:
 I'll try to modify qutebrowser to see it in practice...

**Comment by The-Compiler** [comment]
There's already #842 but I want to have some tests for hints before changing anything, as there are way too many corner cases :wink:

**Comment by The-Compiler** [comment]
Fixed with #1311.
